.github/actions/start-opensearch/action.yml

```diff
@@ -0,0 +1,175 @@
+name: 'Start OpenSearch'
+description: 'Downloads latest build of OpenSearch, installs a plugin, executes and script and then starts OpenSearch on localhost:9200'
+
+inputs:
+  platform:
+    description: 'What platform is this action running on, e.g. "windows-latest" ? '
+    required: true
+
+  opensearch-version:
+    description: 'The version of OpenSearch that should be used, e.g "3.0.0"'
+    required: true
+
+  plugin-name:
+    description: 'The the name of the plugin to use, such as security-dashboards-plugin'
+    required: true
+
+  plugin-start-script:
+    description: 'The file name for the configuration script for the plugin such as install_demo_configurations'
+    required: true
+  # plugins/opensearch-security/tools/install_demo_configuration.sh
+
+  java-version:
+    description: 'The java version to be used, must be either 11 or 17'
+    required: true
+
+  plugin-zip:
+    description: 'The name of the zip file for the plugin hosted on docker-host i.e. security-plugin.zip '
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+
+    # Configure longpath names if on Windows
+    - name: Enable Longpaths if on Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: git config --system core.longpaths true
+      shell: pwsh
+
+    # Downloads the min version of OpenSearch Core from the latest build
+    - name: Download OpenSearch Core for Linux
+      if: ${{ inputs.platform == 'ubuntu-latest' }}
+      run: |
+        wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ inputs.opensearch-version }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ inputs.opensearch-version }}-linux-x64.tar.gz
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+      shell: bash
+
+    - name: Download OpenSearch Core for Windows
+      if: ${{ inputs.platform == 'windows-latest' }}
+      run: |
+        Invoke-WebRequest https://artifacts.opensearch.org/snapshots/core/opensearch/${{ inputs.opensearch-version }}-SNAPSHOT/opensearch-min-${{ inputs.opensearch-version }}-SNAPSHOT-windows-x64-latest.zip -Outfile opensearch-${{ inputs.opensearch-version }}.zip
+        tar -xzf opensearch-${{ inputs.opensearch-version }}.zip
+        del opensearch-${{ inputs.opensearch-version }}.zip
+        Rename-Item opensearch-* opensearch
+      shell: pwsh
+
+    - name: Download OpenSearch Core for MacOS
+      if: ${{ inputs.platform == 'macos-latest' }}
```

Comment:
Looks like this includes steps for mac, but we don't have mac artifacts?  Does it work or is this something that we should roll into another pull request?

Reply:
We do not have mac support yet but since there are no calls for running the runner on Mac I figured it would be easier for whoever adds Mac support if the framework is already there.

Reply:
Please do not include code that is unused.  I've added some lines where folks have the following thoughts on commented-out code, and in this case, I think this code is basically following the same pattern even if it 'could be' active.  When its time to support the mac platform that will be a good time to revisit and update this code with testing.

- [1] https://kentcdodds.com/blog/please-dont-commit-commented-out-code
- [2 ]https://stackoverflow.com/questions/758279/checking-in-of-commented-out-code

Reply:
Alright, I will remove it.